services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/FileHandleManagerImpl.java

```diff
@@ -465,20 +474,41 @@ private String getUrlForGoogleCloudFileHandle(GoogleCloudFileHandle handle) {
 			  We still attempt to override content type because it does not seem to interfere with the call, and
 			  perhaps one day Google may decide to allow us to override content type with this parameter.
 			 */
-			String contentType = handle.getContentType();
-			if (StringUtils.isNotEmpty(contentType) && !NOT_SET.equals(contentType)) {
-				signedUrl += "&response-content-type=" + URLEncoder.encode(contentType, "UTF-8");
+			Optional<String> contentType = getFileHandleResponseContent(handle).get(RESPONSE_CONTENT_TYPE);
+			if (contentType.isPresent()) {
+				signedUrl += "&" + RESPONSE_CONTENT_TYPE + "=" + URLEncoder.encode(contentType.get(), "UTF-8");
 			}
-			String fileName = handle.getFileName();
-			if (StringUtils.isNotEmpty(fileName) && !NOT_SET.equals(fileName)) {
-				signedUrl += "&response-content-disposition=" + URLEncoder.encode(ContentDispositionUtils.getContentDispositionValue(fileName), "UTF-8");
+
+			Optional<String> fileName = getFileHandleResponseContent(handle).get(RESPONSE_CONTENT_DISPOSITION);
+			if (fileName.isPresent()) {
+				signedUrl += "&" + RESPONSE_CONTENT_DISPOSITION + "=" + URLEncoder.encode(ContentDispositionUtils.getContentDispositionValue(fileName.get()), "UTF-8");
 			}
 		} catch (UnsupportedEncodingException e) {
 			throw new RuntimeException("Error encoding query string for signed URL", e);
 		}
 		return signedUrl;
 	}
 
+	private static Map<String, Optional<String>> getFileHandleResponseContent(FileHandle handle) {
+		Map<String, Optional<String>> responseContent = new HashMap<>();
+
+		String contentType = handle.getContentType();
+		if (StringUtils.isNotEmpty(contentType) && !NOT_SET.equals(contentType)) {
+			responseContent.put(RESPONSE_CONTENT_TYPE, Optional.of(contentType));
+		} else {
+			responseContent.put(RESPONSE_CONTENT_TYPE, Optional.empty());
+		}
+
+		String fileName = handle.getFileName();
+		if (StringUtils.isNotEmpty(fileName) && !NOT_SET.equals(fileName)) {
+			responseContent.put(RESPONSE_CONTENT_DISPOSITION, Optional.of(fileName));
```

Comment:
We should use the existing method (ContentDispositionUtils.getContentDispositionValue(fileName))to get response content disposition instead of filename